ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -532,6 +522,10 @@ private void prepareContext() throws CommandProcessorException {
     context.setHDFSCleanup(true);
 
     driverTxnHandler.setContext(context);
+
+    if (SessionState.get() != null) {
+      SessionState.get().addQueryState(getConf().get(HiveConf.ConfVars.HIVEQUERYID.varname), getQueryState());
```

Comment:
Would it be better to use `driverContext.getQueryState().getQueryId()` instead of depending on the conf?

Reply:
I think there is no difference in terms of performance, but I changed it for the sake of readability.